src/NServiceBus.Core/Transports/Msmq/Msmq.cs

```diff
@@ -29,7 +30,7 @@ public MsmqTransport()
         /// <summary>
         /// Configures transport for receiving.
         /// </summary>
-        protected internal override void ConfigureForReceiving(TransportReceivingConfigurationContext context)
+        protected internal override TransportReceivingConfigurationResult ConfigureForReceiving(TransportReceivingConfigurationContext context)
```

Comment:
Why not just change it to?

`TransportReceivingConfigurationContext ConfigureForReceiving();`